packages/react-components/react-table/src/hooks/selectionManager.ts

```diff
@@ -0,0 +1,112 @@
+import { SelectionMode } from './types';
+
+type OnSelectionChangeCallback = (selectedItems: Set<SelectionItemId>) => void;
+
+export interface SelectionManager {
+  toggleSelect(id: SelectionItemId): void;
+  select(id: SelectionItemId): void;
+  deSelect(id: SelectionItemId): void;
+  clearSelection(): void;
+  isSelected(id: SelectionItemId): boolean;
+  toggleSelectAll(itemIds: SelectionItemId[]): void;
```

Comment:
nit: It's kinda internal API and I am a native speaker, but:

- `toggleSelect` (two verbs) => `toggleSelection` / just `toggle` as we have have `select` / `toggleItem`
- `select` -> `selectItem` (check below about `toggleSelectAll`)
- `deSelect` -> `deselect` - there is a word [deselect](https://english.stackexchange.com/questions/18465/unselect-or-deselect)
- `toggleSelectAll` -> `toggleMultiple` -> `toggleItems`?

Reply:
I went for;

- `toggleItem` -> `toggleRow`
- `clearItems` -> `clearRows`
- `selectItem`/`deSelectItem` -> `selectRow`/`deSelectRow`